src/main/scala/org/virtuslab/beholder/filters/json/JsonFilterFieldsComponent.scala

```diff
@@ -91,39 +116,37 @@ trait JsonFilterFieldsComponent extends FilterFieldComponent with SeqParametersH
     /**
      * search in text (ilike) for optional fields
      */
-    object inOptionText extends ImplicitlyJsonFilterFiled[Option[String], String]("OptionalText") {
+    object inOptionText extends SingleFieldJsonFilterFieldFromFormat[Option[String], String]("OptionalText") {
       override def filterOnColumn(column: Rep[Option[String]])(data: String): Rep[Option[Boolean]] = column ilike s"%${escape(data)}%"
     }
 
-    object inDateTime extends ImplicitlyJsonFilterFiled[DateTime, DateTime]("DateTime") {
-      override def filterOnColumn(column: Rep[DateTime])(data: DateTime): Rep[Option[Boolean]] = column.? === data
-
-      override protected[json] def valueWrite: Writes[DateTime] = Writes.jodaDateWrites("yyyy-MM-dd HH:mm")
-
-      override protected def filterFormat: Format[DateTime] = new Format[DateTime] {
-        override def writes(o: DateTime): JsValue = valueWrite.writes(o)
-
-        override def reads(json: JsValue): JsResult[DateTime] = Reads.jodaDateReads("yyyy-MM-dd HH:mm").reads(json)
+    val inDateTime: SingleFieldJsonFilterFieldFromFormat[DateTime, DateTime] = {
+      import play.api.libs.json.JodaReads.jodaDateReads
+      import play.api.libs.json.JodaWrites.jodaDateWrites
```

Comment:
It's imported multiple times, it can stay on top as it was before

Reply:
It's imported twice and those cases differ - in one we define our own format and in other we expect format to be resolved implicitly